response_operations_ui/views/logout.py

```diff
@@ -8,5 +8,7 @@
 @logout_bp.route('/')
 def logout():
     logout_user()
+    if 'messages_survey_selection' in session:
+        session.pop('messages_survey_selection')
```

Comment:
I think a try except block here is still more pythonic. I know @ajmaddaford thinks differently, so won't block the PR on this, but I'd do something like:

```Python
try:
    del(session['messages_survey_selection'])
except KeyError:
    logger.info('No survey to remove from session')
finally:
    flash("You are now signed out", category='successful_signout')
    return redirect(url_for('sign_in_bp.sign_in'))
```

Reply:
- I don't see the value in logging that there is no survey to remove from the session, we're not logging when it is in the session and is removed.
- Having the flash and return in the finally block implies it is in some way linked to the deletion of the survey selection key, which it's not.
- The `session.pop` should be changed to `del(session['messages_survey_selection'])` as the existence of the key has already been checked.

Taking these things into account, I think the options are:
```Python
try:
    del(session['messages_survey_selection'])
except KeyError:
    pass

flash("You are now signed out", category='successful_signout')
return redirect(url_for('sign_in_bp.sign_in'))
```
vs
```Python
if 'messages_survey_selection' in session:
    del(session['messages_survey_selection'])

flash("You are now signed out", category='successful_signout')
return redirect(url_for('sign_in_bp.sign_in'))
```

I'd be happy to go with the first option, I can see there is some clarity from not checking for the existence of the key before deletion.